source/utils/paragraphHelper.py

```diff
@@ -0,0 +1,271 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2022 NV Access Limited, Rob Meredith
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+import api
+import speech
+import controlTypes
+import textInfos
+import tones
+import config
+from NVDAObjects.window.winword import WordDocumentTextInfo
+from NVDAObjects.window.winword import BrowseModeWordDocumentTextInfo
+from NVDAObjects.UIA import UIATextInfo
+from displayModel import EditableTextDisplayModelTextInfo
+from typing import (
+	Tuple,
+	List,
+)
+from enum import IntEnum
+
+MAX_LINES = 250  # give up after searching this many lines
+
+
+class _ErrorTone(IntEnum):
```

Comment:
Would you rather use a dataclass rather than an enum? Semantically, an enum defines a list of values to choose.

Reply:
Yes, that may be a more accurate pattern here, apologies for the misleading suggestion @rob-aph.
I do note that the implementation of a dataclass is more complex

```suggestion
from dataclasses import dataclass

@dataclass(frozen=True)
class _ErrorToneData:
	frequencyHz: int = 1000
	durationMs: int = 30

_ERROR_TONE = _ErrorToneData()
```